server-spi-private/src/main/java/org/keycloak/policy/BlacklistPasswordPolicyProvider.java

```diff
@@ -46,7 +46,7 @@ public PolicyError validate(String username, String password) {
 
     PasswordBlacklist blacklist = (FileBasedPasswordBlacklist) policyConfig;
 
-    if (!blacklist.contains(password.toLowerCase())) {
+    if (!blacklist.contains(password)) {
```

Comment:
If the idea really is to normalize all passwords to lower case befeore comparing to the blacklist, this needs to stay. I just now noticed the comment in KEYCLOAK-11700 introduced in ea248b560183c58f65761053228d343829cf65e2 (" All passwords in the blacklist must be lowercased to facilitate case-insensitive comparison.").

But maybe it would be safer  to force the password to lower-case when reading the blocklist instead of relying on the user to ensure this?

Reply:
I like the idea of lower-casing the values from the blacklist file, which looks like you are already doing.